web/src/DetailedInfoPage.tsx

```diff
@@ -37,21 +37,37 @@ interface SequenceMeta {
 
 // temporary
 interface ParticipantModel {
-    id: number
-    external_id: string
-    reported_sex?: number
-    reported_gender?: string
-    karyotype?: string
-    meta: { [key: string]: any }
+    id: number;
+    external_id: string;
+    reported_sex?: number;
+    reported_gender?: string;
+    karyotype?: string;
+    meta: { [key: string]: any };
 }
 
 interface SampleSequencing {
-    id: number
-    external_ids?: { [name: string]: string }
-    sample_id: string
-    type: string
-    meta?: SequenceMeta
-    status: string
+    id: number;
+    external_ids?: { [name: string]: string };
+    sample_id: string;
+    type: string;
+    meta?: SequenceMeta;
+    status: string;
+}
+
+enum SampleType {
+    BLOOD = "blood",
+    SALIVA = "saliva",
+}
+
+interface Sample {
+    id: number | string;
+    external_id: string;
+    participant_id?: number;
+    active?: boolean;
+    meta?: { [key: string]: any };
+    type?: SampleType;
+    project?: number;
+    author?: string;
 }
 
 const sampleFieldsToDisplay = ["active", "type", "participant_id"];
```

Comment:
I think it would be worth turning `DetailedInfoPage` into a state component, then we can use a [React error boundary](https://reactjs.org/docs/error-boundaries.html) to show better error messages and not have the page crash.

Even just display the JSON response if we have one to the user, as that might be useful if there are display issues.

Reply:
Not sure what you mean by a state component. 
Some of the changes in this PR should fix a lot of crashes.

Reply:
Sorry I meant a class component, then you can implement these two methods to bound any further exceptions:

```typescript
class DetailedInfoPage extends React.Component {

  static getDerivedStateFromError(error) {
    // Update state so the next render will show the fallback UI.
    return { hasError: true };
  }

  componentDidCatch(error, errorInfo) {
    // You can also log the error to an error reporting service
    logErrorToMyService(error, errorInfo);
  }

  render() {
    if (this.state.hasError) {
      // You can render any custom fallback UI
      return <h1>Something went wrong.</h1>;
    }

    return this.props.children; 
  }

Reply:
I was under the impression class components should never be used these days. But admittedly I don't know much about them. 
There are some npm packages which seem to mimic the behaviour (https://www.npmjs.com/package/react-use-error-boundary) if they are of interest. 
Happy to discuss on Monday :) (Think it's wise to ask @daniaki for some expertise as he knows more React than I do)